content/js1/sprints/2/prep/index.md

```diff
@@ -8,4 +8,406 @@ backlog= 'Module-JS1'
+
+{{<note type="activity" title="exercise">}}
+Given the code above, try using the visualisation diagrams to work out what happens when `makeGreeting` is called on line 8.
+
+What happens to the arrow indicating the current line of code that we're executing.
+What else do you notice when `makeGreeting` is called above?
+
+{{</note>}}
+
+The arrow representing the current line of execution now moves to the first line inside the function: this shows that the computer now starts running instructions inside the function.
+
+In addition, another frame is created when we call `makeGreeting`:
+
+![make-greeting-frame](make-greeting-frame.png)
+
+Like the global frame, the `makeGreeting` frame will be used to track variables that are created when `makeGreeting` is executed.
+
+### 📥 Returning from a function
+
+Earlier we defined output for the function by writing a return statement and setting a return value.
+
+<iframe width="800" height="350" frameborder="0" src="https://pythontutor.com/iframe-embed.html#code=function%20makeGreeting%28%29%20%7B%0A%20%20%20%20const%20greeting%20%3D%20%60Hello%20there%60%3B%0A%20%20%20%20return%20greeting%3B%0A%7D%0A%0Aconst%20firstName%20%3D%20'German'%3B%0Aconst%20lastName%20%3D%20'Bencii'%3B%0A%0A%0A%0AmakeGreeting%28%29%3B&codeDivHeight=400&codeDivWidth=350&cumulative=false&curInstr=0&heapPrimitives=nevernest&origin=opt-frontend.js&py=js&rawInputLstJSON=%5B%5D&textReferences=false"> </iframe>
+
+However, we can see that the `return` keyword also has additional effects. As well as defining the output, it also tells us to "return" to the place in our code where the function was called.
+
+{{<note type="activity" title="exercise">}}
+Try using the visualisation diagram above to work out what happens when the computer executes the return statement.
+
+{{</note>}}
+
+### 🧩 Putting everything together
+
+So now we have some idea of how the computer executes our code, we can apply these ideas to a program for our current implementation of `convertToPercentage`:
+
+<iframe width="800" height="350" frameborder="0" src="https://pythontutor.com/iframe-embed.html#code=function%20convertToPercentage%28%29%20%7B%0A%20%20const%20num%20%3D%200.5%3B%0A%20%20const%20percentage%20%3D%20%60%24%7Bnum%20*%20100%7D%25%60%3B%0A%20%20return%20percentage%3B%0A%7D%0A%0Aconst%20output1%20%3D%20convertToPercentage%280.5%29%3B%0Aconst%20output2%20%3D%20convertToPercentage%280.3%29%3B&codeDivHeight=400&codeDivWidth=350&cumulative=false&curInstr=0&heapPrimitives=nevernest&origin=opt-frontend.js&py=js&rawInputLstJSON=%5B%5D&textReferences=false"> </iframe>
+
+{{<note type="activity" title="exercise">}}
+
+Earlier we noted that when the computer executes the code above, we get the following output:
+
+```bash
+50%
+50%
+```
+
+Given the code above, try using the visualisation diagrams to "play computer" with this example. Try stepping through each line of code to explain why the program is giving this output
```

Comment:
We haven't defined what playing computer is before telling our trainees to "play computer" - I think we should add an explicit definition above, along the lines of "read through the program line be line, saying out loud what each value we see is, and what the computer will do next"